### DISCUSSION TITLE
It'd be nice to integrate Nextra with an existing NextJS project

### DISCUSSION THREAD

**Discussion by corysimmons** [shuding/nextra#93] [Ideas]
Right now, if you add a `docs` dir and even add a `path: '/docs'` option to `theme.config.js`, it will still list every dir/file within `pages` in the table of contents:

<img src="https://user-images.githubusercontent.com/224636/107574734-044bce80-6bbd-11eb-9c22-c24ece701340.png" width="200">

It'd be nice if I could just add a `pages/docs` dir, then change `path: '/docs'` in `theme.config.js`, and be able to populate `pages/docs` with all kinds of MDX files. 🤷

**Comment by tbecerrp** [comment]
Hi, I can't implementation nextra with productive app with next.  
I want something like this: 
```
-pages
        -Docs
            -doc.md
        -_app.tsx
```

**Comment by maltejur** [comment]
Next.js recommends to just use a different next.js app for that and then add rewrites https://nextjs.org/docs/advanced-features/multi-zones

**Reply by corysimmons** [reply]
That works! Thanks @maltejur

**Reply by IlirBajrami** [reply]
How multizones work with app directory routing?

**Comment by mikhaelesa** [comment]
I tried this but it doesn't work. When i tried to access the docs page the page is blank but the title of the docs is showing.

It also gives me this error `Invalid href passed to next/router: \themes\docs\configuration, repeated forward-slashes (//) or backslashes \ are not valid in the href`.

Please share your workaround on this. It might be a great help since there is not much documentation about this. Thanks

**Reply by ishafiul** [reply]
I am also facing the same issue. and now it shows 
```
Error: Hydration failed because the initial UI does not match what was rendered on the server. 
```
please let me know if you have any solution for this

**Comment by DatoKhojava** [comment]
hello guys, I just want to have a nextjs 14 app with other stuff (login,logout,api key generator,profile generator) and also documentation, I already have login logout working stack with custom rest api, is it possible to integrate nextra to my existing project ?

**Reply by corysimmons** [reply]
I think Nextra just works with the `pages` dir, but I think you can use both the `pages` dir and `app` dir.

Ultimately whatever auth solution you used probably has methods and/or endpoints that work with either.

```bash
- app
  - almost_all_of_your_app
  - auth
    - routes.ts
- pages
  - nextra_docs # use getServerSideProps to hit your auth endpoint
```

**Reply by DatoKhojava** [reply]
first of all thank you for reply, I decided to fork nextra repo and add my custom methods and components and I think it will work, yesterday I just wanted to integrate nextra to already existing app. I think better to fork and add stuff

anyway thank you
